### PR TITLE
mention feature removed, support for image added 

### PR DESCRIPTION
**Pull request by arpandhakal**

### CODE REVIEW

**Review by aatbip — CHANGES_REQUESTED** [review]
Is this the final work for the image upload? I think the resize dragger that we want for the task app is a bit different than what we have here. Image loader and pdf uploads are also seems not working.

File: lib/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx [hunk 1]
```diff
@@ -1,72 +1,72 @@
...
-          tiptapEditorUtils.deleteRange(range)
-          tiptapEditorUtils.toggleHeading(2)
+          const tiptapEditorUtils = new TiptapEditorUtils(editor);
+          tiptapEditorUtils.deleteRange(range);
+          tiptapEditorUtils.toggleHeading(2);
         },
       },
       {
-        title: 'Heading 3',
+        title: "Heading 3",
```
Comment (rrojan) [inline lib/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx]: Change prettier to prefer single quotes instead of double... this is causing a lot of unnecessary changes

File: lib/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx [hunk 2]
```diff
@@ -1,72 +1,72 @@
...
       {
-        title: 'Text',
+        title: "Text",
         command: ({ editor, range }: { editor: Editor; range: any }) => {
-          const tiptapEditorUtils = new TiptapEditorUtils(editor)
-          tiptapEditorUtils.deleteRange(range)
-          tiptapEditorUtils.setParagraph()
+          const tiptapEditorUtils = new TiptapEditorUtils(editor);
+          tiptapEditorUtils.deleteRange(range);
+          tiptapEditorUtils.setParagraph();
```
Comment (rrojan) [inline lib/components/tiptap/floatingMenu/floatingMenuSuggestion.tsx]: Same with semi. Please set `"semi": false` in your prettier config

File: lib/components/tiptap/image/image.ts [hunk 3]
```diff
@@ -1,103 +1,164 @@
...
+      deleteImage: undefined,
+    };
+  },
+  addCommands() {
+    const { deleteImage } = this.options;
+    return {
+      setImage:
+        (options) =>
+        ({ commands }) => {
+          const uniqueId = `image-${Date.now()}-${Math.random()
```
Comment (rrojan) [inline lib/components/tiptap/image/image.ts:50]: Could we instead do `actualFilename-${Date.now()}` only?

You might have to split user's original filename with `.` and discard the last item of array -> join rest to get actualFilename

Comment (rrojan) [inline lib/components/tiptap/image/image.ts]: If you use `Date.now()` the random part shouldn't be unnecessary right? Or you could just use the random part. 

Is this function for displaying an image or uploading one?

File: lib/components/tiptap/image/image.ts [hunk 4]
```diff
@@ -1,103 +1,164 @@
...
+                state.tr.replaceWith(
+                  selection.from,
+                  selection.to,
+                  state.schema.nodes.paragraph.create()
+                )
+              );
+
+            if (deleteImage) {
+              deleteImage(imageId);
+            }
```
Comment (rrojan) [inline lib/components/tiptap/image/image.ts]:
```suggestion
              deleteImage?.(imageId);
```

**Review by rrojan — CHANGES_REQUESTED** [review]

File: lib/components/tiptap/image/ImageResizeComponent.tsx [hunk 5]
```diff
@@ -1,39 +1,102 @@
...
+    const handleUpload = async () => {
+      if (props.editor) {
+        const { handleImageUpload } =
+          props.editor.options.extensions.find(
+            (ext: any) => ext.name === "imageResize"
+          )?.options || {};
+        if (handleImageUpload) {
+          try {
+            await handleImageUpload(id);
+            setLoading(false);
```
Comment (rrojan) [inline lib/components/tiptap/image/ImageResizeComponent.tsx]: Can you confirm that if `handleImageUpload` is not provided then there is no way for setLoading to get stuck at `true` (since setLoading(false) is inside if block

**Review by rrojan — CHANGES_REQUESTED** [review]

File: lib/components/tiptap/image/image.ts (hunk 3 above)
Comment (rrojan) [reply lib/components/tiptap/image/image.ts:68]: shouldn't*

File: src/main.tsx [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { ImagePickerUtils } from '../lib/utils/imagePickerUtils.ts'
 
 const App = () => {
   const [content, setContent] = useState<string>(
-    '<p> ashdkasd </p> <img src = "https://picsum.photos/200/300" width="100%" height="auto" /> <p> hello </p>'
+    '<p> ashdkasd </p> <img src = "https://picsum.photos/200/300" width ="75" height="112" /> <p> hello </p>'
```
Comment (rrojan) [inline src/main.tsx:8]: ??

Is this an app to test the Tapwrite package? Why is this being pushed to Github?

**Review by rrojan — CHANGES_REQUESTED** [review]
